From 647a7761e2fa423c6e1bd6785b043dbe7b525e3c Mon Sep 17 00:00:00 2001
From: Lukas Nykryn <lnykryn@redhat.com>
Date: Thu, 25 Jun 2015 09:20:59 +0200
Subject: [PATCH] Revert "core: one step back again, for nspawn we actually
 can't wait for cgroups running empty since systemd will get exactly zero
 notifications about it"

This reverts commit 743970d2ea6d08aa7c7bff8220f6b7702f2b1db7.

RHEL-only
https://bugzilla.redhat.com/show_bug.cgi?id=1141137
https://github.com/systemd/systemd/pull/350

Resolves: #1199644
---
 src/core/unit.c | 10 +---------
 1 file changed, 1 insertion(+), 9 deletions(-)

diff --git a/src/core/unit.c b/src/core/unit.c
index b9e1f13ea..fa17567dd 100644
--- a/src/core/unit.c
+++ b/src/core/unit.c
@@ -3546,15 +3546,7 @@ int unit_kill_context(
                                 log_unit_warning_errno(u->id, r, "Failed to kill control group: %m");
                 } else if (r > 0) {
 
-                        /* FIXME: For now, we will not wait for the
-                         * cgroup members to die, simply because
-                         * cgroup notification is unreliable. It
-                         * doesn't work at all in containers, and
-                         * outside of containers it can be confused
-                         * easily by leaving directories in the
-                         * cgroup. */
-
-                        /* wait_for_exit = true; */
+                        wait_for_exit = true;
 
                         if (c->send_sighup && k != KILL_KILL) {
                                 set_free(pid_set);
